src/test/scala/pt/tecnico/dsi/neutron/CrudSpec.scala

```diff
@@ -0,0 +1,68 @@
+package pt.tecnico.dsi.neutron
+
+import cats.effect.IO
+import cats.implicits._
+import io.circe.{Decoder, Encoder}
+import pt.tecnico.dsi.neutron.models.Model
+import pt.tecnico.dsi.neutron.services.{BulkCreate, CrudService}
+import pt.tecnico.dsi.openstack.common.models.WithId
+
+abstract class CrudSpec[T <: Model](val name: String, val service: NeutronClient[IO] => CrudService[IO, T])
+  (implicit val encoder: Encoder[T#Create], val decoder: Decoder[WithId[T#Read]])
+  extends Utils {
+
+  val displayName: String = name.capitalize
+  val createStub: IO[T#Create]
+
+  val withSubCreated: IO[(WithId[T#Read], CrudService[IO, T])] =
+    for {
+      neutron <- client
+      stub <- createStub
+      crudService = service(neutron)
+      createdStub <- crudService.create(stub)
+    } yield (createdStub, crudService)
+
+
+  s"$displayName service" should {
+
+    "create and get" in {
+      for {
+        (stub, service) <- withSubCreated
+        actual <- service.get(stub.id)
+      } yield actual shouldBe stub
+    }
+
```

Comment:
Suggestion:
```scala
"create and get" in withSubCreated.flatMap { (stub, service) =>
  service.get(stub.id).idempotently(_ shouldBe stub)
}
```
I like it because the part that is actually the test is indented to be inside the test.